django-stubs/contrib/auth/models.pyi

```diff
@@ -12,9 +12,20 @@ from django.db.models.manager import EmptyManager
 from django.utils.functional import _StrOrPromise
 from typing_extensions import Self
 
-_AnyUser: TypeAlias = Model | AnonymousUser
+# This is our "placeholder" type the mypy plugin refines to configured 'AUTH_USER_MODEL'
+# wherever it is used as a type. The most recognised example of this is (probably)
+# `HttpRequest.user`
+_User: TypeAlias = AbstractBaseUser
 
-def update_last_login(sender: type[AbstractBaseUser], user: AbstractBaseUser, **kwargs: Any) -> None: ...
+_AnyUser: TypeAlias = _User | AnonymousUser
```

Comment:
I would move `UserModel` here, since here we define all the aliases and then re-export it where needed.

Reply:
Django doesn't declare a `UserModel` here so if you'd prefer this I would suggest we declare it as `_UserModel` and when re-exporting it (assuming you want that) we'd use `from django.contrib.auth.models import _UserModel as UserModel`